src/stedi/cdk/lambda.clj

```diff
@@ -1,37 +1,28 @@
 (ns stedi.cdk.lambda
-  (:require [clojure.string :as string]
+  (:require [clojure.java.io :as io]
+            [clojure.string :as string]
             [stedi.cdk :as cdk]
-            [stedi.cdk.lambda.build :as build]))
+            [stedi.lambda.build :as lambda-build]))
 
 (cdk/import ["@aws-cdk/core" Duration]
-            ["@aws-cdk/aws-lambda" Function AssetCode Runtime LayerVersion])
+            ["@aws-cdk/aws-lambda" Code Runtime Function])
 
-(defn Clj
-  "Wraps `@aws-cdk/aws-clambda.Function` to be clojure friendly.
-
-    fn          - A var pointing to a handler function.
-    environment - A map of environment variables
-    handler     - Only necessary if fn isn't specified
-    aot         - A list of namespaces to AOT compile
-  "
-  [parent id {:keys [fn environment handler aot] :as props}]
-  (let [{:keys [lib-layer src]} (build/build aot)
-
-        env       (merge environment
-                         (when fn {"STEDI_LAMBDA_HANDLER" (-> fn symbol str)}))
-        function  (Function parent id
-                            (merge {:code        (AssetCode src)
-                                    :handler     (or handler "stedi.lambda.entrypoint::handler")
-                                    :runtime     (:JAVA_8 Runtime)
-                                    :environment env
-                                    :memorySize  2048
-                                    :timeout     (Duration/minutes 1)}
-                                   (dissoc props :environment :aot :fn)))
-        lib-layer (LayerVersion function "lib-layer" {:code (AssetCode lib-layer)})]
-    (Function/addLayers function lib-layer)
-    function))
-
-(defn ^:deprecated clj
-  "Deprecated in favor of `stedi.cdk.lambda/Clj`."
-  [_ parent id {:keys [fn environment handler aot] :as props}]
-  (Clj parent id props))
+(defn fn-from-var
+  "Creates a @aws-cdk/aws-lambda.Function from a Clojure var. Includes
+  sane defaults that can be overridden by passing in optional
+  `function-props`."
```

Comment:
For those of us less informed, either a link or elaboration to what `function-props` are would be appreciated, e.g., "For supported function props, see https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda.FunctionProps.html"

Reply:
Will fix